### DISCUSSION TITLE
Unable to access organization private repository with fine-grained token

### DISCUSSION THREAD

**Discussion by LeviPesin** [community/community#40910] [Apps, API and Webhooks] [answered]
### Select Topic Area

Bug

### Body

It is currently impossible to access a private repository belonging to an organization by using a fine-grained token belonging to an organization member (even the owner), even if the fine-grained token includes `contents: write` permissions and has "All repositories" option enabled.

Tested with `curl https://api.github.com/repos/ORG/REPO -H "Authorization: Bearer github_pat_OWNER_TOKEN"`, returns `{
  "message": "Not Found",
  "documentation_url": "https://docs.github.com/rest/reference/repos#get-a-repos
itory"
}
`. Works perfectly when using a classic PAT token.

Proposed solution: add a new option "All repositories of an organization" to the repositories' selection?

**Comment by byrneh** [comment, marked as answer]
Each token can only access resources owned by a single user or organization.
Each token can only access specific repositories
When creating the FG-PAT 

Under Resource owner, select the resource as. The target org orgnaiztion.
The token will only be able to access resources owned by the selected resource owner. Organizations that you are a member of will not appear unless the organization opted in to fine-grained personal access tokens

**Reply by LeviPesin** [reply]
> Under Resource owner, select the resource as. The target org orgnaiztion.

There is no such option there. The only option that I see is @LeviPesin, and there is not an option @warriors-life.

**Reply by byrneh** [reply]
Hi @LeviPesin see last sentence of my previous reply. 
Organizations that you are a member of **will not appear unless the organization opted in to fine-grained personal access tokens**.

**Reply by LeviPesin** [reply]
Thank you, with the opting in it seems to work!

**Reply by dvargasfuertes** [reply]
> The token will only be able to access resources owned by the selected resource owner

Is it possible for the token to have access to resources that the selected resource owner _has access to_ but is _not owned_ by the resource owner? My desired end state is for my token to have my (dvargas92495) as the resource owner but for it to still have access to private repos owned by my organization (samepage-network)

**Reply by hughj095** [reply]
Organizational repos require classic PATs AND authorization of Org SSO.  This can be done after creating the token--select Authorize SSO in the token.  Go to your picture profile -> Settings -> Developer Settings -> Personal Access Tokens (classic) -> create a token.

**Reply by Radon8472** [reply]
> Organizational repos require classic PATs

Do you mean, there is currently NO WAY to access organisation repositories via fine graned PAT, and the only option is to use classic tokens and grant access to ALL organisations I belong to ??

**Reply by pgunold-evc** [reply]
> Under Resource owner, select the resource as. The target org orgnaiztion. The token will only be able to access resources owned by the selected resource owner. Organizations that you are a member of will not appear unless the organization opted in to fine-grained personal access tokens

My organisation opted in to fine-grained personal access tokens, but I cant grand `repo:write` or `repo:read` permissions, there is no such an option in the permissions select field.

**Comment by mahtisshop** [comment]
https://github.com/orgs/community/discussions/40910#discussioncomment-9104551

**Reply by mahtisshop** [reply]
iهر توکن فقط می تواند به منابع متعلق به یک کاربر یا سازمان دسترسی داشته باش
هر توکن
هنگام ایجاد FG-PAT فقط می تواند به مخازن خاصی دسترسی داشته باشد

در قسمت مالک منبع، منبع را به عنوان انتخاب کنید. سازمان هدف.
توکن فقط می‌تواند به منابع متعلق به مالک منبع انتخابی دسترسی داشته باشد. سازمان‌هایی که شما عضو آنها هستید ظاهر نمی‌شوند مگر اینکه سازمان از نشانه‌های دسترسی شخصی دقیق استفاده کند
